src/components/store/ingredients.ts

```diff
@@ -0,0 +1,57 @@
+import { StateCreator } from "zustand";
+import { NewIngredient } from "../types.ts";
+
+interface MyIngredientsActions {
+  addIngredient: (ingredient: NewIngredient) => void;
+  editIngredients: (ingredients: NewIngredient[]) => void;
+  editIngredientsBoughtOften: (
+    ingredients: {
+      id: NewIngredient["id"];
+      amount: number;
+      frequency: number;
+    }[]
+  ) => void;
+}
+
+interface MyIngredientsState {
+  ingredients: NewIngredient[];
+
+  ingredientsBoughtOften: {
+    id: NewIngredient["id"];
+    amount: number;
+    frequency: number;
+  }[];
+}
+
+const initialIngredientsState: MyIngredientsState = {
```

Comment:
Vytahnuti initial stavu ven, to se nam v budoucnu muze hodit, treba kdyz to budem nejak sofistikovane resetovat.